src/main/scala/mesosphere/marathon/core/event/impl/stream/HttpEventStreamServlet.scala

```diff
@@ -27,16 +27,17 @@ class HttpEventSSEHandle(request: HttpServletRequest, emitter: Emitter) extends
 
   private val subscribedEventTypes = request.getParameterMap.getOrDefault("event_type", Array.empty).toSet
 
-  def subscribed(eventType: String): Boolean = {
-    subscribedEventTypes.isEmpty || subscribedEventTypes.contains(eventType)
-  }
+  def subscribedForEventType(eventType: String): Boolean = subscribedEventTypes.isEmpty || subscribedEventTypes.contains(eventType)
+  def subscribedForTaskId(taskId: String): Boolean       = subscribedEventTypes.isEmpty || subscribedEventTypes.contains(task_id)
 
   override def remoteAddress: String = request.getRemoteAddr
 
   override def close(): Unit = emitter.close()
 
   override def sendEvent(event: MarathonEvent): Unit = {
-    if (subscribed(event.eventType)) blocking(emitter.event(event.eventType, event.jsonString))
+  	if subscribedForEventType(event.eventType)) && subscribedForTaskId(event.taskId)) blocking(emitter.event(event.eventType, event.jsonString))
+  	if subscribedForEventType(event.eventType)) blocking(emitter.event(event.eventType, event.jsonString))
+	if subscribedForTaskId(event.taskId)) blocking(emitter.event(event.taskId, event.jsonString))
```

Comment:
It seems that `event_type` and `task_id` are `OR`ed, not `AND`ed. Is it as you want it to be? Could you please explain why? As far as I understand, the purpose of filtering is to reduce the number of events received, therefore `AND` logic would be more appropriate here.